interbtc-spec/docs/source/spec/nomination.rst

```diff
@@ -208,463 +131,154 @@ Specification
-
-
-.. _executeOperatorWithdrawal:
-
-executeOperatorWithdrawal
--------------------------
-
-Execute all matured (unbonded) withdrawal requests of an operator.
-
-Specification
-.............
-
-*Function Signature*
-
-``executeOperatorWithdrawal(operatorId)``
-
-*Parameters*
-
-* ``operatorId``: the id of the requester.
-
-*Events*
-
-* ``ExecuteOperatorCollateralWithdrawal(operatorId, unbondedCollateral)``
+* ``DepositCollateral(vaultId, nominatorId, amount)``
 
-*Errors*
+*Preconditions*
 
-* ``VaultNotOptedInToNomination``: the ``operatorId`` is not an Operator.
+* The BTC Parachain status in the :ref:`security` component MUST NOT be ``SHUTDOWN:2``.
+* A vault with id ``vaultId`` MUST be registered.
+* A vault with id ``vaultId`` MUST have nomination enabled.
 
-* ``NoMaturedCollateral``: either no collateral withdrawal has been requested, or the requests have not matured yet.
-
-Preconditions
-.............
-
-* The BTC Parachain status in the :ref:`security` component must be set to ``RUNNING:0``.
-
-
-Function Sequence
-.................
-
-1. Check ``operatorId`` is an operator. If not, throw ``VaultNotOptedInToNomination``.
-
-2. Iterate through the ``withdrawalRequests`` in the ``Operator`` struct to determine how much collateral was unbonded, removing matured requests.
-
-3. If there is zero unbonded collateral, throw ``NoMaturedCollateral``.
-
-
-.. _executeNominatorWithdrawal:
-
-executeNominatorWithdrawal
---------------------------
-
-Execute all matured (unbonded) withdrawal requests of a nominator.
-
-Specification
-.............
-
-*Function Signature*
-
-``executeNominatorWithdrawal(nominatorId, operatorId)``
-
-*Parameters*
-
-* ``nominatorId``: the id of the requester.
-
-* ``operatorId``: the id of the operator.
-
-*Events*
-
-* ``ExecuteNominatorCollateralWithdrawal(nominatorId, operatorId, unbondedCollateral)``
-
-*Errors*
-
-* ``VaultNotOptedInToNomination``: the ``operatorId`` is not an Operator.
-
-* ``NoMaturedCollateral``: either no collateral withdrawal has been requested, or the requests have not matured yet.
-
-* ``NominatorNotFound``: the ``nominatorId`` is not a Nominator.
-
-Preconditions
-.............
-
-* The BTC Parachain status in the :ref:`security` component must be set to ``RUNNING:0``.
-
-
-Function Sequence
-.................
-
-1. Check ``operatorId`` is an operator. If not, throw ``VaultNotOptedInToNomination``.
-
-2. Check ``nominatorId`` is a nominator. If not, throw ``NominatorNotFound``.
-
-3. Iterate through the ``withdrawalRequests`` array in the ``Nominator`` struct inside the ``Operator`` struct for ``operatorId``. Determine how much collateral was unbonded, removing matured requests.
-
-4. If there is zero unbonded collateral, throw ``NoMaturedCollateral``.
+*Postconditions*
 
+* The vault's collateral MUST increase by the amount nominated.
+* The nominators MUST earn rewards equivalent to their collateral.
```

Comment:
This is vague - better to be concrete: which storage values changes to make this happen? Describe the stake change

Reply:
Missing postcondition: nominator's free balance decreases by `amount`